hbase-common/src/main/java/org/apache/hadoop/hbase/io/encoding/BufferedDataBlockEncoder.java

```diff
@@ -954,25 +1033,55 @@ private int compareTypeBytes(Cell key, Cell right) {
       return 0;
     }
 
+    // These findCommonPrefix* methods rely on the fact that keyOnlyKv is the "right" cell argument
```

Comment:
Is it possible to enforce the type of the right Cell? So later when someone changes the implementation of upper layer, we will know that we should also change the implementation of this method

Reply:
Good call! In https://github.com/apache/hbase/pull/5347/commits/458e99f78c6b6cce383b92abec9568fff496ffae, I've changed the type of the right cell to be `KeyValue.KeyOnlyKeyValue` s.t. if the upper layer is refactored the author will be forced to address the resulting type errors. Let me know if you had something more comprehensive in mind here @Apache9 !